src/server/frontend_xwayland/xcb_connection.h

```diff
@@ -244,6 +244,7 @@ class XCBConnection
     Atom const clipboard;
     Atom const clipboard_manager;
     Atom const utf8_string;
+    Atom const compound_text{"COMPOUND_TEXT", this};
```

Comment:
Can you move the initialization to the `XCBConnection` constructor for consistency with the rest of the atoms? I thought there was a reason they had to be constructed there. If not, we might want to move all the constructors to the header for easier maintenance, but we should be consistent about it.